### PR TITLE
deps.json : add nanovg download

### PR DESCRIPTION
**Pull request by vinsentli**

### CODE REVIEW

**Comment by vinsentli** [comment]
> Please change the URL to download from the upstream and not from your personal repo.

The original NanoVG demo relies on OpenGL and GLFW APIs to compile successfully. So i forked the original NanoVG repository, and made some minor modifications to the demo.c file.
Here are the changes I made：https://github.com/vinsentli/nanovg/commit/5660d826c3bde672f6d92fde502c06d3029273b7

Alternatively, you can fork a new repository from the "igl" branch of my repository？
https://github.com/vinsentli/nanovg/tree/igl

**Comment by corporateshark** [comment]
> > Please change the URL to download from the upstream and not from your personal repo.
> 
> The original NanoVG demo relies on OpenGL and GLFW APIs to compile successfully. So i forked the original NanoVG repository, and made some minor modifications to the demo.c file. Here are the changes I made：[vinsentli/nanovg@5660d82](https://github.com/vinsentli/nanovg/commit/5660d826c3bde672f6d92fde502c06d3029273b7)
> 
> Alternatively, you can fork a new repository from the "igl" branch of my repository？ https://github.com/vinsentli/nanovg/tree/igl

IGL already has GLFW and OpenGL, so no problem with that. Alternatively, the bootstrapping script can apply a small patch locally if you make it.

**Comment by vinsentli** [comment]
> > > Please change the URL to download from the upstream and not from your personal repo.
> > 
> > 
> > The original NanoVG demo relies on OpenGL and GLFW APIs to compile successfully. So i forked the original NanoVG repository, and made some minor modifications to the demo.c file. Here are the changes I made：[vinsentli/nanovg@5660d82](https://github.com/vinsentli/nanovg/commit/5660d826c3bde672f6d92fde502c06d3029273b7)
> > Alternatively, you can fork a new repository from the "igl" branch of my repository？ https://github.com/vinsentli/nanovg/tree/igl
> 
> IGL already has GLFW and OpenGL, so no problem with that. Alternatively, the bootstrapping script can apply a small patch locally if you make it.

done.
I have made a nanovg.patch

File: third-party/deps/patches/nanovg.patch [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+-#include <GLFW/glfw3.h>
++//#include <GLFW/glfw3.h>
+ #include "nanovg.h"
+ 
+ #ifdef _MSC_VER
+@@ -41,12 +42,15 @@ void startGPUTimer(GPUtimer* timer)
+ 
+ void startGPUTimer(GPUtimer* timer)
+ {
++#if 0
```
[the remark this answers is not in the document]
Comment (vinsentli) [reply third-party/deps/patches/nanovg.patch:51]: Let me think again.

**Comment by vinsentli** [comment]
perf.c is used for show FPS.
Clipboard_Screenshot_1732957059

File: third-party/deps/patches/nanovg.patch [hunk 2]
```diff
@@ -0,0 +1,76 @@
+diff --exclude=.git --exclude=.hg -rupN example/demo.c example_new/demo.c
+--- example/demo.c	2024-11-28 20:44:24
++++ example_new/demo.c	2024-11-25 11:41:56
+@@ -5,9 +5,9 @@
+ #ifdef NANOVG_GLEW
+ #  include <GL/glew.h>
+ #endif
+-#include <GLFW/glfw3.h>
++//#include <GLFW/glfw3.h>
```
[the remark this answers is not in the document]
Comment (vinsentli) [reply third-party/deps/patches/nanovg.patch]: done

**Review by corporateshark — APPROVED** [review]

**Pull request closed without merging**